Sources/Basics/FileSystem/FileSystem+Extensions.swift

```diff
@@ -302,10 +302,6 @@ extension FileSystem {
     public func writeFileContents(_ path: AbsolutePath, string: String) throws {
         return try self.writeFileContents(path, bytes: .init(encodingAsUTF8: string))
     }
-
-    public func writeFileContents(_ path: AbsolutePath, provider: () -> String) throws {
-        return try self.writeFileContents(path, body: { stream in stream <<< provider() })
```

Comment:
This version that was using a stream operator created all parent directories of `path` if those didn't exist, which is inconsistent with the other overloads. This version feels quite redundant to `writeFileContents(_ path: AbsolutePath, string: String)`, so I've cleaned it up and adjusted tests to create parent directories when needed.